### PR TITLE
Merge data layer

### PR DESCRIPTION
**Pull request by AaronNewman1495**
Need to update documentation

### CODE REVIEW

File: inventory/models/request.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import user
+from item import *
+
+class Request(db.Model):
+	id = db.Column(db.Integer, primary_key=True)
+	description = db.Column(db.String())
+	user_id = db.Column(db.Integer, db.ForeignKey('user.id'))
+	item_id = db.Column(db.Integer, db.ForeignKey('item.id'))
+	quantity = db.Column(db.Integer)
+	status = db.Column(db.String())
```
Comment (danrili) [inline inventory/models/request.py]: Make smallint to discretize statuses

File: inventory/models/tag.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	item_id = db.Column(db.Integer, db.ForeignKey('item.id'))
+
+	def __init__(self, text, item_name):
+		self.text = text
+		self.addTag(item_name)
+		db.session.add(self)
+		db.session.commit()
+
+	def addTag(self, item_name):
+		item = Item.query.filter_by(name=item_name).first()
```
Comment (danrili) [inline inventory/models/tag.py]: internally, use id to refer to objects

File: inventory/models/user.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+from inventory import db
+import request
+
+class User(db.Model):
+	id = db.Column(db.Integer, primary_key=True)
+	first_name = db.Column(db.String())
+	last_name = db.Column(db.String())
+	net_id = db.Column(db.String(), unique=True)
```
Comment (danrili) [inline inventory/models/user.py]: maybe use varchar?

File: inventory/models/user.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	role = db.Column(db.String())
```
Comment (danrili) [inline inventory/models/user.py]: role should be a separate table. this is so we can handle permissions

File: inventory/models/request.py (hunk 1 above)
Comment (AaronNewman1495) [inline inventory/models/request.py]: Done

File: inventory/models/tag.py (hunk 2 above)
Comment (AaronNewman1495) [inline inventory/models/tag.py]: Done

File: inventory/models/user.py (hunk 4 above)
Comment (AaronNewman1495) [inline inventory/models/user.py]: Done

File: inventory/models/user.py [hunk 5]
```diff
@@ -6,24 +6,41 @@ class User(db.Model):
 	first_name = db.Column(db.String())
 	last_name = db.Column(db.String())
 	net_id = db.Column(db.String(), unique=True)
-	role = db.Column(db.String())
 	requests = db.relationship('Request', backref='user', lazy='dynamic')
+	type = db.Column(db.String())
```
Comment (danrili) [inline inventory/models/user.py]: What is type? Is that role? Isn't that supposed to be another table now.

File: inventory/models/user.py [hunk 6]
```diff
@@ -6,24 +6,41 @@ class User(db.Model):
...
 
-	def __init__(self, first_name, last_name, net_id, role):
+	__mapper_args__ = {
+		'polymorphic_on':type,
+		'polymorphic_identity':'employee'
+	}
+
+	def __init__(self, first_name, last_name, net_id):
```
Comment (danrili) [inline inventory/models/user.py]: The user should still be instantiated with a role. It'd be something like a roleId. That would be a foreign key reference to the id on the role table.

I image for the role table, we can just have columns id, name (str), permissions (jsonb). permissions would be a json map of permissions.

File: inventory/models/user.py (hunk 5 above)
Comment (AaronNewman1495) [inline inventory/models/user.py]: Fixed

File: inventory/models/user.py (hunk 6 above)
Comment (AaronNewman1495) [inline inventory/models/user.py]: Done

**Comment by danrili** [comment]
:shipit:

**Comment by danrili** [comment]
🚢 ship it

**Pull request merged**